lib/graceful-exit.js

```diff
@@ -148,8 +150,22 @@ exports.middleware = function middleware (app) {
 
   return function checkIfExitingGracefully (req, res, next) {
     if (app.settings.graceful_exit === true) {
-      // sorry keep-alive connections, but we need to part ways
-      req.connection.setTimeout(1);
+      if (options.respondDuringExit) {
+        logger('Server unavailable, incoming request rejected with error: ' + req && req.headers || '?');
+        // Signal the caller that the connection is closing, if before suicideTimeout
+        // 2019-09-10 - HH - Fix request processing after connection closes, see issue 14
+        res.set('Connection', 'close');
+        return next(
+          options.unavailableError() ||
+          defaultOptions.unavailableError(
+              new Error('Server unavailable, no new requests accepted during shutdown')
+            )
+        );
+      } else {
+        logger('Server unavailable, incoming request dropped silently: ' + req && req.headers || '?');
```

Comment:
ditto here, I think you'll get `[Object object]` in the logs because of the `+` operator converting headers to string